### PR TITLE
Saving multiple 'high beam' detected frames in the working directory

### PR DESCRIPTION
**Pull request by samdish**

### CODE REVIEW

**Comment by samdish** [comment]
Added the required changes.

File: Core.cpp [hunk 1]
```diff
@@ -45,10 +48,18 @@ int main(int, char**)
 
 		if (bboxes.size() && !flag) {
 			//High Beam detected
+
+			detected_image = frame; //save detected frame;
```
Comment (sakshamsaxena) [inline Core.cpp:57]: The `frame` here is cropped etc. Not what we need to save. `detected_image` should ideally be assigned after line 43.

File: Core.cpp [hunk 2]
```diff
@@ -45,10 +48,18 @@ int main(int, char**)
...
+
+			sprintf(file,"Image_%d.jpg",count);  //save files with different names using 'count'
```
Comment (sakshamsaxena) [inline Core.cpp]: `file` isn't defined anywhere I think. Also, C++ has direct string concatenation with `+` operator. Can probably write this as `string file = "Image_"+to_string(count)+".jpg";`

File: Core.cpp [hunk 3]
```diff
@@ -45,10 +48,18 @@ int main(int, char**)
 
 		if (bboxes.size() && !flag) {
 			//High Beam detected
+
+			detected_image = frame; //save detected frame;
+
+			sprintf(file,"Image_%d.jpg",count);  //save files with different names using 'count'     
+			imwrite(file,image);
```
Comment (sakshamsaxena) [inline Core.cpp]: `detected_image` not `image` bro

File: Core.cpp (hunk 2 above)
Comment (samdish) [reply Core.cpp]: Went through a few links on stack overflow, and found that sprintf is the standard that everyone follows.. Can experiment with '+' as well.

File: Core.cpp (hunk 3 above)
Comment (samdish) [reply Core.cpp]: Oh ya! But you get what I meant: p

File: Core.cpp (hunk 1 above)
Comment (samdish) [reply Core.cpp:57]: By using it after line 43, we'll end up saving every frame that we capture.

Comment (sakshamsaxena) [reply Core.cpp:57]: We'll only save it when we enter the `if` block, that is, only on detection. It'll be the original unaltered frame. We only need to keep a copy of every unaltered frame, and it'll only be saved on detection. Right now, you've frame, original_frame, detected_frame. We could instead simply do `detected_frame = frame` above and and then use that directly in `imwrite`.

Comment (sakshamsaxena) [reply Core.cpp:57]: Reconsider the flow man. You'll get it.

**Review by sakshamsaxena — CHANGES_REQUESTED** [review]

**Review by sakshamsaxena — APPROVED** [review]

**Pull request merged**